test/hotspot/jtreg/runtime/os/TestMemoryAllocationLogging.java

```diff
@@ -0,0 +1,247 @@
+                        String.format("Released \\[0x.* - 0x.*\\), \\(%d bytes\\)", PAGE_SIZE)
+                };
+                break;
+            }
+            case "testAttemptedReserveFailed": {
+                expectedLogs = new String[] {
+                        /* Debug level log */
+                        String.format("Reserved \\[0x.* - 0x.*\\), \\(%d bytes\\)", PAGE_SIZE),
+                        String.format("Attempt to reserve \\[0x.* - 0x.*\\), \\(.* bytes\\) failed"),
+                        /* Trace level log */
+                        "mmap failed: \\[0x.* - 0x.*\\), \\(.* bytes\\) errno=\\(Not enough space\\)"
+                };
+                break;
+            }
+            case "testReserveFailed": {
+                expectedLogs = new String[] {
+                        /* Debug level log */
+                        "Reserve failed \\(.* bytes\\)",
+                        /* Trace level log */
+                        "mmap failed: \\[0x.* - 0x.*\\), \\(.* bytes\\) errno=\\(Not enough space\\)"
+                };
+                break;
+            }
+            case "testCommitFailed": {
+                expectedLogs = new String[] {
+                        /* Debug level log */
+                        String.format("Failed to commit \\[0x.* - 0x.*\\), \\(%d bytes\\)", COMMIT_SIZE),
+                        /* Trace level log */
+                        "mmap failed: \\[0x.* - 0x.*\\), \\(.* bytes\\) errno=\\(Invalid argument\\)"
+                };
+                break;
+            }
+            case "testUncommitFailed": {
+                expectedLogs = new String[] {
+                        /* Debug level log */
+                        String.format("Reserved \\[0x.* - 0x.*\\), \\(%d bytes\\)", PAGE_SIZE),
+                        "Failed to uncommit \\[0x.* - 0x.*\\), \\(.* bytes\\)",
+                        /* Trace level log */
+                        "mmap failed: \\[0x.* - 0x.*\\), \\(.* bytes\\) errno=\\(Not enough space\\)"
+                };
+                break;
+            }
+            case "testReleaseFailed": {
+                expectedLogs = new String[] {
+                        /* Debug level log */
+                        "Failed to release \\[0x.* - 0x.*\\), \\(.* bytes\\)",
+                        /* Trace level log */
+                        "munmap failed: \\[0x.* - 0x.*\\), \\(.* bytes\\) errno=\\(Invalid argument\\)"
+                };
+                break;
+            }
+
+            default: {
+                throw new RuntimeException("Invalid test " + args[0]);
+            }
+        }
+
+        OutputAnalyzer output = runTestWithOptions(options);
+        checkExpectedLogMessages(output, expectedLogs);
+    }
+
+    private static OutputAnalyzer runTestWithOptions(String[] options) throws IOException {
+        ProcessBuilder pb = ProcessTools.createLimitedTestJavaProcessBuilder(options);
```

Comment:
If you are using the "limited" test process then I think the overall test should specify `@requires vm.flagless`.